src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntry.cs

```diff
@@ -44,5 +46,21 @@ public override string ToString()
         {
             return $"Path: {Path}, VR:{VR}, Level:{Level} Status:{Status}";
         }
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(Path, VR, Level.GetHashCode(), Status.GetHashCode());
+        }
+
+        public override bool Equals(object obj)
+        {
+            CustomTagEntry other = obj as CustomTagEntry;
+            if (other == null)
+            {
+                return false;
+            }
+
+            return Path == other.Path && VR == other.VR && Level == other.Level && Status == other.Status;
```

Comment:
> Path == other.Path && VR == other.VR [](start = 18, length = 37)

Are `Path` and `VR` case-sensitive?

Reply:
yes, nice catch! will make a case-insensitive comparison.

Reply:
If `Path` and `VR` are `null`-able, then you can use `string.Equals` instead

---
In reply to: [567147794](https://github.com/microsoft/dicom-server/pull/501#discussion_r567147794) [](ancestors = 567147794)